scripts/update-tsconfig-references.ts

```diff
@@ -0,0 +1,97 @@
+import * as shell from 'shelljs';
+
+type PackageJSON = {
+    dependencies?: { [name: string ]: string };
+    devDependencies?: { [name: string ]: string };
+};
+
+type TSConfig = {
+    references?: { path: string }[];
+};
+
+const getReferencesFromDependencies = (packagePath: string): string[] => {
+    const packageJSONPath = `${packagePath}/package.json`;
+    const packageJSON: PackageJSON = require(`../../${packageJSONPath}`);
+
+    const dependencies = Object.keys(packageJSON.dependencies || {});
+    const devDependencies = Object.keys(packageJSON.devDependencies || {});
+    const allDependencies = dependencies.concat(devDependencies);
+
+    // Map `hint` and `@hint/*` dependencies from `package.json` to `tsconfig.json` reference names.
+    return allDependencies
+        .filter((name) => {
+            return name === 'hint' || name.startsWith('@hint/');
+        })
+        .map((name) => {
+            return name.replace('@hint/', '');
+        });
+};
+
+const compactReferences = (json: string): string => {
+    // Condense JSON-serialized references to fit on a single line.
+    return json.replace(/\{\r?\n\s*("path": "[^"]+")\r?\n\s*}/g, '{ $1 }');
+};
+
+const updateFile = (filePath: string, content: string) => {
+    const writeContent = (shell as any)['ShellString']; // eslint-disable-line dot-notation
+
+    writeContent(content).to(filePath);
+};
+
+const main = () => {
+
+    // Include all sub-packages (except configurations and connector-edge)
+    const subPackages = shell.ls('packages')
+        .filter((packageName: string) => {
+            return !packageName.startsWith('configuration-') && packageName !== 'connector-edge';
```

Comment:
Why are you ignoring `connector-edge`?

Reply:
Because it only builds on Windows (and even then we don't typically build it as part of the full build when doing local development). It still builds individually.

Reply:
ok! :)